07-routing-gossip.md

```diff
@@ -559,6 +563,11 @@ the channel when the peer reestablishes contact.  Because gossip
 messages are batched and replace previous ones, the result may be a
 single seemingly-redundant update.
 
+The simple `splicing` flag warns nodes that this channel will be
+replaced; it tells them to give a grace period for the new channel to
+appear (it can't be announced until it's 6 blocks deep) during which
+it can use the old channel id.
```

Comment:
Do we really need that? Non-upgraded nodes won't understand this new flag, which is quite a downside.

What about *always* giving a grace period to channel spends? This is something we could in theory deploy today, so the network would already have that by the time we actually deploy splicing software.

Or we could detect that the channel spend actually creates another output with the same `scriptPubKey` and thus infer that it's not a closing tx? Is there a downside to re-using the same funding pubkeys across splices?

Reply:
Yes, this should be a separate PR which proposes everyone delay pruning by 12(?) blocks.

Don't want to duplicate keys, because unannounced channels still want stealth.

Reply:
Made a PR encapsulating this change https://github.com/lightning/bolts/pull/1004

Reply:
Ok looks like this PR need a rebase because https://github.com/lightning/bolts/pull/1004 got merged